ui/components/app/modals/confirm-delete-network/confirm-delete-network.component.js

```diff
@@ -1,46 +1,28 @@
 import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
+import { toEvmCaipChainId } from '@metamask/multichain-network-controller';
+
 import Modal, { ModalContent } from '../../modal';
 
 export default class ConfirmDeleteNetwork extends PureComponent {
   static propTypes = {
     hideModal: PropTypes.func.isRequired,
     removeNetwork: PropTypes.func.isRequired,
     onConfirm: PropTypes.func.isRequired,
-    switchEvmNetwork: PropTypes.func.isRequired,
     networkNickname: PropTypes.string.isRequired,
     chainId: PropTypes.string.isRequired,
-    currentChainId: PropTypes.string.isRequired,
-    ethereumMainnetClientId: PropTypes.string.isRequired,
   };
 
   static contextTypes = {
     t: PropTypes.func,
   };
 
   handleDelete = async () => {
-    const {
-      chainId,
-      currentChainId,
-      ethereumMainnetClientId,
-      onConfirm,
-      hideModal,
-      removeNetwork,
-      switchEvmNetwork,
-    } = this.props;
-
-    // An implicit auto-switch in the network-controller should occur
-    // to mainnet when the network being deleted is the current selected
-    // EVM network and the active network is a non-EVM network.
-    //
-    // TODO: This logic must be ported to the
-    // multichain-network-controller so the "remove use case" can
-    // be properly handled.
-    if (chainId === currentChainId) {
-      await switchEvmNetwork(ethereumMainnetClientId);
-    }
+    const { chainId, onConfirm, hideModal, removeNetwork } = this.props;
 
-    await removeNetwork(chainId);
+    // NOTE: We can only support EVM networks, so the conversion is safe here.
```

Comment:
Sorry, my wording was bad with my previous suggestion: 

```suggestion
    // NOTE: We only support EVM networks removal, so the conversion is safe here.
```